x/poolmanager/keeper.go

```diff
@@ -106,3 +107,29 @@ func (k Keeper) SetNextPoolId(ctx sdk.Context, poolId uint64) {
 func (k *Keeper) SetPoolIncentivesKeeper(poolIncentivesKeeper types.PoolIncentivesKeeperI) {
 	k.poolIncentivesKeeper = poolIncentivesKeeper
 }
+
+// GetPool gets the pool based on poolId.
+func (k *Keeper) GetPool(ctx sdk.Context, poolId uint64) (types.PoolI, error) {
+	swapModule, err := k.GetPoolModule(ctx, poolId)
+	if err != nil {
+		return nil, err
+	}
+
+	pool, poolErr := swapModule.GetPool(ctx, poolId)
+	if poolErr != nil {
+		return nil, poolErr
+	}
+
+	return pool, nil
+}
```

Comment:
There's already a `RoutePool` method that does exactly the same. Let's use that to avoid duplication